In [9]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import chi2

## Question 8

Investigate following mortality patterns

| week | chinese | jewish |
|------|---------|--------|
| -2   | 55      | 141    |
| -1   | 33      | 145    |
| 1    | 70      | 139    |
| 2    | 49      | 161    |

Table shows number of old women who died in the weeks before and after the Chinese Harvest Moon Festival

In [3]:
data = np.array([
    [55, 141],
    [33, 145],
    [70, 139],
    [49, 161]
])

### Test independence between race and week of death using chi2

In [4]:
chi2_contingency(data)

(12.420794151882923,
 0.006072304928267781,
 3,
 array([[ 51.16267339, 144.83732661],
        [ 46.46406053, 131.53593947],
        [ 54.55611602, 154.44388398],
        [ 54.81715006, 155.18284994]]))

p-value of 0.006 shows there is strong evidence that that race and week of death are dependent

### test that number of deaths for jewish people is independent of week

Let pi = probability of death in each week. H0: pi = 0.25 for all i

In [13]:
chinese_deaths = data[:,0]
chinese_deaths

array([55, 33, 70, 49])

In [14]:
chinese_deaths_expected = np.full((4,), 0.25*chinese_deaths.sum())
chinese_deaths_expected

array([51.75, 51.75, 51.75, 51.75])

In [22]:
test_statistic = (
    (chinese_deaths - chinese_deaths_expected) ** 2 / chinese_deaths_expected
).sum()

test_statistic

13.579710144927537

In [23]:
p_value = 1-chi2.cdf(test_statistic, df=3)
p_value

0.0035368204078958243

p-value of 0.003 is strong evidence that probability of death for chinese women depends on week number.

### test that number of deaths for jewish people is independent of week

Let pi = probability of death in each week. H0: pi = 0.25 for all i

In [24]:
jewish_deaths = data[:,1]
jewish_deaths

array([141, 145, 139, 161])

In [25]:
jewish_deaths_expected = np.full((4,), 0.25*jewish_deaths.sum())
jewish_deaths_expected

array([146.5, 146.5, 146.5, 146.5])

In [26]:
test_statistic = (
    (jewish_deaths - jewish_deaths_expected) ** 2 / jewish_deaths_expected
).sum()

test_statistic

2.040955631399317

In [27]:
p_value = 1-chi2.cdf(test_statistic, df=3)
p_value

0.5639498711784423

p-value of 0.56 is evidence that number of deaths for jewish women is independent of week number